### PR TITLE
feat: implement reset button and test for edge cases

### PR DESCRIPTION
**Pull request by HarryMumford**

### CODE REVIEW

File: src/components/Board/Board.jsx [hunk 1]
```diff
@@ -51,6 +53,14 @@ class Board extends Component {
     this.isRunning = false
   }
 
+  reset () {
+    this.pause()
+    this.board.reset()
+    this.board = new BoardLogic(this.emptyBoard(), CellLogic)
+    this.setState({ generationCount: this.board.getGenerationCount() })
```
Comment (Hyan18) [inline src/components/Board/Board.jsx]: you can set the state of generationCount and cells in one call of setState

**Comment by ffgi-es** [comment]
Already merged due to Jamie's inability to branch off from master.
